### PR TITLE
Add warnings about potential misalignment and propagate call site.

### PR DESCRIPTION
**Pull request by rainwoodman**

### CODE REVIEW

**Comment by rainwoodman** [comment]
@sbird

File: mpsort-mpi.c [hunk 1]
```diff
@@ -224,31 +259,17 @@ _create_segment_group(struct SegmentGroupDescr * descr, size_t * sizes, size_t *
...
-        size_t val;
-        int   rank;
-    } leader_st;
-
-    leader_st.val = sizes[ThisTask];
-    leader_st.rank = rank;
+    /* rank with most data in a group is the leader of the group. */
+    descr->group_leader_rank = MPIU_GetLoc(&sizes[ThisTask], MPI_LONG, MPI_MAX, descr->Group);
 
-    MPI_Allreduce(MPI_IN_PLACE, &leader_st, 1, MPI_LONG_INT, MPI_MAXLOC, descr->Group);
```
Comment (sbird) [inline mpsort-mpi.c]: Hang on, isn't the code before just wrong? sizeof(leader_st) == 12 != sizeof(MPI_LONG_INT) == 8?

**Comment by rainwoodman** [comment]
Long Int was a struct of long and int. Unlike LongLong which is the same as
Long on most platforms. If I get the quirks right. ;)


Yu

**Comment by sbird** [comment]
I see! I think this is probably a good idea: MPI_LONG_INT with Allreduce may indeed be buggy in a number MPI impl. But I don't think misalignment is the issue here: I am compiling with -Zp which forces struct alignment to 8 bytes, and one may manually verify that the radix size used in MP-Gadget is always a factor of 8.

**Comment by sbird** [comment]
I would like to copy mpsort from an external dependency of MP-Gadget to an in-tree component, to make it easier to work on, reduce duplicate code and allow the test to run in CI. Do you want to merge this first?

**Comment by rainwoodman** [comment]
I do use an internal mpsort in FastPM. There I usually keep a local clone of mpsort, and do cp /path/to/mpsort/*.[ch] to the source directory when I want to get an update.

**Pull request merged**